core/templates/dev/head/pages/collection_player/CollectionPlayer.js

```diff
@@ -108,6 +120,81 @@ oppia.controller('CollectionPlayer', [
         explorationId).getExplorationSummaryObject();
     };
 
+    $scope.generatePathParameters = function() {
+      var pathSvgParameters = 'M250 80  C 500 100, 500 280, 250 300';
+      var collectionNodeCount = $scope.collection.getCollectionNodeCount();
+      var sParameterExtension = '';
+      $scope.svgHeight = 220;
+      if (collectionNodeCount === 1) {
+        $scope.pathIconParameters = $scope.generatePathIconParameters();
+        return '';
+      } else if (collectionNodeCount === 2) {
+        $scope.pathIconParameters = $scope.generatePathIconParameters();
+        return pathSvgParameters;
+      } else {
+        var y = 500;
+        for (var i = 1; i < Math.floor(collectionNodeCount / 2); i++) {
+          if (i % 2 === 0) {
+            x = 500;
+          } else {
+            x = 0;
+          }
+          sParameterExtension += x + ' ' + y + ', ';
+          y += 20;
+          sParameterExtension += 250 + ' ' + y + ', ';
+          y += 200;
+        }
+        pathSvgParameters += ' S ' + sParameterExtension;
+      }
+      if (collectionNodeCount % 2 === 0) {
+        $scope.svgHeight = y - 280;
+      } else {
+        $scope.svgHeight = y - 150;
+      }
+      $scope.pathIconParameters = $scope.generatePathIconParameters();
+      return pathSvgParameters;
+    };
+
+    $scope.generatePathIconParameters = function() {
+      var collectionNodes = $scope.collection.getCollectionNodes();
+      var arr = [];
+      arr.push({
+          thumbnailIconUrl:
+            collectionNodes[0].getExplorationSummaryObject().thumbnail_icon_url,
+          left: '225px',
+          top: '60px',
+          thumbnailBgColor:
+            collectionNodes[0].getExplorationSummaryObject().thumbnail_bg_color
+        });
+      var x = 225;
```

Comment:
Same comments above apply -- what do the numbers mean, what does `count` represent (maybe give it a more descriptive name), etc.